### PR TITLE
feat: Implemented a language dropdown in the summary section

### PR DESCRIPTION
**Pull request by Bandinikhil**
feat: 

- Implemented a language dropdown in the summary section 
- And updated API, translations to support Hindi and Odia for dynamic content, including titles, ministries, summaries, questions, and sub-questions.

### CODE REVIEW

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts [hunk 1]
```diff
@@ -181,6 +182,43 @@ export class ConsultationQuestionnaireComponent
...
+
+    if (this.currentLanguage === 'or') {
+      const questionTextOdia = question?.odiaQuestionText;
+      if (questionTextOdia) {
+        return questionTextOdia;
+      }
+    }
+
+    return question?.questionText;
+  }
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts:203]: Similar logic is being repeated multiple times. Move the logic to a shared function and pass variable data as arguments

File: src/app/modules/consultations/consultation-profile/consultation-profile.graphql.ts [hunk 2]
```diff
@@ -444,12 +487,16 @@ export const SubmitResponseQuery = gql`
...
+                    hindiQuestionText
+                    odiaQuestionText
                     questionType
                     supportsOther
                     isOptional
                     subQuestions {
                       id
                       questionText
+                      hindiQuestionText
+                      odiaQuestionText
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/consultation-profile.graphql.ts:499]: Move duplicated parts of queries to fragments where possible

File: src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.html [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+          bindValue="id"
+          [clearable]="false"
+          [searchable]="false"
+          (change)="onLanguageChange()"
+          [(ngModel)]="selectedLanguage">
+        </ng-select>
+      </div>
+    </div>
+    <div *ngIf="profileData?.englishSummary || profileData?.hindiSummary || profileData?.odiaSummary">
+      <app-page-content-renderer [page]="useSummaryOdia ? profileData?.odiaSummary : useSummaryHindi ? profileData?.hindiSummary : profileData?.englishSummary">
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.html:21]: Move this logic to the `component.ts` file. Create `get` functions to compute and return these values

File: src/app/shared/services/translation.service.ts [hunk 4]
```diff
@@ -67,57 +67,60 @@ setIndex() {
...
+
+  const length = this.dictionary.value.length;
+
+  if (this.currentLanguage === 'hi') {
+      return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, 'translation') || text;
+  } else if (this.currentLanguage === 'or') {
+      return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, 'translation2') || text;
+  } else {
+      return text;
+  }
```
Comment (gpalsingh) [inline src/app/shared/services/translation.service.ts:82]:
```suggestion

let tranlsationKey;

  if (this.currentLanguage === 'hi') {
      translationKey = 'translation';
  } else if (this.currentLanguage === 'or') {
      translationKey = 'translation2';
  } else {
      return text;
  }
  
  return this.binarySearch(this.dictionary.value, text.toLowerCase(), 0, length, translationKey) || text;
```

Avoid repeating code where possible

File: src/app/shared/services/translation.service.ts [hunk 5]
```diff
@@ -67,57 +67,60 @@ setIndex() {
...
-    if (midText === text) {
-        return arr[mid].translation;
-    }
-
-    if (midText > text) {
-        return this.binarySearch(arr, text, start, mid - 1);
-    } else {
-        return this.binarySearch(arr, text, mid + 1, end);
-    }
+binarySearch(arr: Array<Dictionary>, text: string, start: number, end: number, field: string) {
```
Comment (gpalsingh) [inline src/app/shared/services/translation.service.ts:85]: This code is wrongly formatted. The rest of the file is using indents of 4 spaces

File: src/app/shared/services/translation.service.ts [hunk 6]
```diff
@@ -67,57 +67,60 @@ setIndex() {
...
+  }
+
+  if (midText > text) {
+      return this.binarySearch(arr, text, start, mid - 1, field);
+  } else {
+      return this.binarySearch(arr, text, mid + 1, end, field);
+  }
 }
 
+
```
Comment (gpalsingh) [inline src/app/shared/services/translation.service.ts:123]: Remove unnecessary changes like this

**Review by gpalsingh — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts [hunk 7]
```diff
@@ -25,7 +25,6 @@ export class AppComponent implements OnInit, OnDestroy {
   languages = [
     { id: "en", name: "English" },
     { id: "hi", name: "Hindi" },
-    { id: "or", name: "Odia" }
```
Comment (gpalsingh) [inline src/app/app.component.ts]: If this is intentional and not temporary then code related to this option should also be removed

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts [hunk 8]
```diff
@@ -183,42 +185,19 @@ export class ConsultationQuestionnaireComponent
...
-
-    if (currentLanguage === 'or' && subQuestion.odiaQuestionText) {
-      return subQuestion.odiaQuestionText;
-    }
-
-    return subQuestion.questionText;
+    return getTranslatedText(this.currentLanguage, {
+      hindi: subQuestion?.hindiQuestionText,
+      odia: subQuestion?.odiaQuestionText
+    }, subQuestion?.questionText);
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts:198]: Both of these functions are the same, just the argument name is different

File: src/app/modules/consultations/consultation-profile/profile-card/profile-card.component.ts [hunk 9]
```diff
@@ -29,11 +30,15 @@ export class ProfileCardComponent implements OnInit, OnChanges {
...
-              private router: Router
-              ){}
+
+  constructor(
+    private consultationsService: ConsultationsService,
+    private userService: UserService,
+    private cookieService: CookieService,
+    private router: Router
+  ) {
+    this.currentLanguage = this.cookieService.get("civisLang");
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/profile-card/profile-card.component.ts:40]: Why is this in `constructor` and not in `ngOnInit`?

File: src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts [hunk 10]
```diff
@@ -163,28 +163,23 @@ export class ReadRespondComponent implements OnInit {
 
   setLanguage() {
     this._cookieService.put('civisLang', this.selectedLanguage);
-    this.getProfileSummary();
+    // this.getProfileSummary();
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts:166]: Remove commented code

**Review by gpalsingh — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 7 above)
Comment (Bandinikhil) [reply src/app/app.component.ts]: @gpalsingh - We're in the process of adding Odia as the next feature, but this option has been temporarily disabled.

File: src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts [hunk 11]
```diff
@@ -184,18 +184,19 @@ export class ConsultationQuestionnaireComponent
...
-      hindi: question?.hindiQuestionText,
-      odia: question?.odiaQuestionText
-    }, question?.questionText);
+      hindi: isSubQuestion ? item?.hindiQuestionText : item?.hindiQuestionText,
+      odia: isSubQuestion ? item?.odiaQuestionText : item?.odiaQuestionText
+    }, isSubQuestion ? item?.questionText : item?.questionText);
+  }
+
+  getQuestionText(question) {
+    return this.getTranslatedTextForQuestion(question);
```
Comment (gpalsingh) [inline src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts:195]: `isSubQuestion` is unnecessary. The result is always same

**Review by gpalsingh — CHANGES_REQUESTED** [review]

**Review by gpalsingh — APPROVED** [review]
